File: objectiveai-sdk-js/src/functions/inventions/state/http.ts
import { ObjectiveAI, type RequestOptions } from "../../../client";
import type { RemotePathCommitOptional } from "../../../remotePathCommitOptional";
import type { FunctionsInventionsStateGetFunctionInventionStateResponse } from "./getFunctionInventionStateResponse";

export function functionsInventionsStateGetFunctionInventionState(
  client: ObjectiveAI,
  params: RemotePathCommitOptional,
  options?: RequestOptions,
): Promise<FunctionsInventionsStateGetFunctionInventionStateResponse> {
  return client.post_unary<FunctionsInventionsStateGetFunctionInventionStateResponse>("functions/inventions/state", params, options);
}

File: objectiveai-sdk-py/objectiveai_sdk/functions/inventions/state/http.py
"""HTTP functions for invention state endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai_sdk.client import ObjectiveAI
    from objectiveai_sdk.functions.inventions.state import GetFunctionInventionStateResponse
    from objectiveai_sdk.remote_path_commit_optional import RemotePathCommitOptional


async def get_function_invention_state(
    client: ObjectiveAI, params: RemotePathCommitOptional,
) -> GetFunctionInventionStateResponse:
    """Retrieve a function invention state definition."""
    return await client.post_unary("functions/inventions/state", params)
